mantidproject/mantid
# D2B detector efficiency calculation

This PR extends the `PowderDiffILLDetEffCorr` algorithm to support the efficiency calculation for D2B instrument, which is a scanning powder diffractometer at the ILL. 

Some satellite algorithms are also fixed/adjusted correspondingly; there were minor bug fixes and extensions.

**To test:**
Use the files 532008 and 532009 from system test data under ILL/D2B directory (set facility to ILL) : 
```
PowderDiffILLDetEffCorr(CalibrationRun='532008,532009', DerivationMethod='GlobalSummedReference2D', OutputWorkspace='calib')
```
The output should be a 2D map of detector efficiencies, similar to the plot in the documentation.

Play around with the input properties of the algorithm, make sure the behaviour is reasonable. Also try to break it.

Make sure the documentation reads clear and the testing is adequate.

The method and the results are preliminarily agreed upon with the instrument scientist.



Fixes #21550 . 
Fixes #22093 .
Refs #21729 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
